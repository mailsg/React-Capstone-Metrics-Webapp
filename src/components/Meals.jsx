import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link, useLocation } from 'react-router-dom';
import Card from 'react-bootstrap/Card';
import { fetchMeals, resetSearchQuery } from '../redux/slice/meals/mealsSlice';
import SearchBox from './SearchBox';
import '../App.css';

const Meals = () => {
  const dispatch = useDispatch();
  const mealsList = useSelector((state) => state.meals.meals);
  const searchQuery = useSelector((state) => state.meals.searchQuery);
  const location = useLocation();

  useEffect(() => {
    dispatch(fetchMeals());
  }, [dispatch]);

  useEffect(() => {
    if (location.pathname === '/') {
      dispatch(resetSearchQuery());
    }
  }, [location.pathname, dispatch]);

  const filteredMeals = mealsList.filter(
    (meal) => meal.strCategory.toLowerCase().includes(searchQuery.toLowerCase()),
  );

  const renderMeals = searchQuery ? filteredMeals : mealsList;

  return (
    <div className="meals-container">
      <SearchBox />
      {renderMeals.map((meal) => (
        <Card
          key={meal.idCategory}
          className="meal-card"
        >
          <Link to={`/about/${meal.idCategory}`} className="meal-link">

            <Card.Img variant="top" src={meal.strCategoryThumb} />
            <Card.Body>
              <Card.Title>{meal.strCategory}</Card.Title>
            </Card.Body>
          </Link>
        </Card>
      ))}
    </div>
  );
};

export default Meals;
