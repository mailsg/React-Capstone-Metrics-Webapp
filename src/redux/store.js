import { configureStore } from '@reduxjs/toolkit';
import mealsReducer from './slice/meals/mealsSlice';

const store = configureStore({
  reducer: {
    meals: mealsReducer,
  },
});

export default store;
