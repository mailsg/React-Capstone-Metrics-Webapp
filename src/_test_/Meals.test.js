import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import Meals from '../components/Meals';

describe('Meals', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router> 
            <Meals />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
