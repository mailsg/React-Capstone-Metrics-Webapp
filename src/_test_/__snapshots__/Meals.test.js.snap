// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meals renders correctly 1`] = `
<div
  className="meals-container"
>
  <div
    className="search-box"
  >
    <input
      onChange={[Function]}
      placeholder="Search meals.."
      type="text"
      value=""
    />
  </div>
</div>
`;
